moveit_ros/planning_interface/move_group_interface/src/move_group.cpp

```diff
@@ -155,44 +193,35 @@ class MoveGroup::MoveGroupImpl
   }
 
   template<typename T>
-  void waitForAction(const T &action, const ros::Duration &wait_for_server, const std::string &name)
+  void waitForAction(const T &action, const std::string &name, const ros::WallTime &timeout, double allotted_time)
   {
     ROS_DEBUG("Waiting for MoveGroup action server (%s)...", name.c_str());
 
-    // in case ROS time is published, wait for the time data to arrive
-    ros::Time start_time = ros::Time::now();
-    while (start_time == ros::Time::now())
-    {
-      ros::WallDuration(0.001).sleep();
-      // explicit ros::spinOnce on the callback queue used by NodeHandle that manages the action client
-      ( ( ros::CallbackQueue * ) node_handle_.getCallbackQueue())->callAvailable();
-    }
-
     // wait for the server (and spin as needed)
-    if (wait_for_server == ros::Duration(0, 0))
+    if (timeout == ros::WallTime())
     {
       while (node_handle_.ok() && !action->isServerConnected())
       {
-        ros::WallDuration(0.001).sleep();
+        ros::WallDuration(0.1).sleep();
```

Comment:
This is a clear use-case for a comment explaining how the magic number was set, something like:

"From testing, 0.001 has been found to capture the connection time expected from the callback queue"